client/chunk.js

```diff
@@ -32,9 +32,11 @@ var Chunk = function (position_, blocks_, colors_) {
   }
 
   this.getRandomSpawnPosition = function(){
-    if(validSpawnLocations == []){
+    // If this is the first call for this chunk, populate the valid spots
+    if(validSpawnLocations.length == 0){
```

Comment:
we don't want to recalculate this every time... something to fix in the future

Reply:
(in the case that there really aren't any valid spawn locations)

Reply:
maybe store a variable like, spawnlocationscalculated = false and then set it to true once you've done the operation once

Reply:
This value is stored in the chunk and will be populated after the frist time it runs for the chunk.

Reply:
Added boolean to track if it needs to be calculated.